src/Controller.sol

```diff
@@ -0,0 +1,220 @@
+contract Controlled
+{
+	bool private initialized = false;
+	address public owner;
+	Controller internal controller;
+
+	modifier whitelistOnly
+	{
+		controller.assertIsWhitelisted(msg.sender);
+		_;
+	}
+
+	modifier controllerOnly
+	{
+		require(Controller(msg.sender) == controller);
+		_;
+	}
+
+	modifier ownerOnly
+	{
+		require(msg.sender == owner);
+		_;
+	}
+
+	modifier onlyInGoodTimes
+	{
+		controller.stopInEmergency();
+		_;
+	}
+
+	modifier onlyInBadTimes
+	{
+		controller.onlyInEmergency();
+		_;
+	}
+
+	function Controlled()
+	{
+		owner = msg.sender;
+	}
+
+	// TODO: convert into constructor call
+	function initialize(Controller _controller) external ownerOnly returns(bool)
+	{
+		require(initialized == false);
+		initialized = true;
+		controller = _controller;
+		owner = address(0);
+		return true;
+	}
+
+	function setController(Controller _controller) public controllerOnly returns(bool)
+	{
+		controller = _controller;
+		return true;
+	}
+
+	function suicideFunds(address _target) public controllerOnly returns(bool)
+	{
+		selfdestruct(_target);
+		return true;
+	}
+}
+
+contract Controller
+{
+	address public owner;
+	mapping(address => bool) public whitelist;
+	mapping(bytes32 => address) public registry;
+	bool public stopped = false;
+
+	modifier whitelistOnly
+	{
+		assertIsWhitelisted(msg.sender);
+		_;
+	}
+
+	modifier devModeOwnerOnly
+	{
+		require(msg.sender == owner);
+		require(whitelist[owner]);
+		_;
+	}
+
+	modifier ownerOnly
+	{
+		require(msg.sender == owner);
+		_;
+	}
+
+	modifier onlyInBadTimes
+	{
```

Comment:
The original names were:
`stopInEmergency` and `onlyInEmergency`.  I always found that `stopInEmergency` caused me to pause when reading and have to think for a moment, which is when I came up with `onlyInGoodTimes` as a replacement that was (hopefully) easier to read without breaking pace.  However, I wanted the inverse modifier to be named in a similar way, rather than having a seemingly unrelated modifier name, which is when I came up with `onlyInBadTimes`.

I'm not convinced that `onlyInGoodTimes` and `onlyInBadTimes` are great names, though I would like to come up with a naming convention that is easy to read in stride and where the two modifiers are clearly paired.

Perhaps someone knows an antonym for emergency?

I also considered `onlyWhenStarted`, but the system defaults to `started` state which makes it a bit confusing.  I also considered `onlyWhenRunning`, but that begs the question of "when what is running".

Another option that OpenZeppelin uses is `Pausable` with `onlyWhenPaused` and `onlyWhenUnpaused`.  Once again though, this doesn't imply a _system wide_ outage, it instead implies that a single contract is paused for some reason.  In this case, the state of emergency is system wide and I was hoping to come up with a good name that encapsulates that somehow (hence the usage of "times").